src/map/attack.cpp

```diff
@@ -108,33 +108,23 @@ void CAttack::SetCritical(bool value)
             }
         }
 
-        SKILLTYPE skilltype = SKILLTYPE::SKILL_NONE;
+        SKILLTYPE skilltype  = SKILLTYPE::SKILL_NONE;
+        SLOTTYPE  weaponSlot = static_cast<SLOTTYPE>(GetWeaponSlot());
 
         if (m_attacker->objtype == TYPE_PC)
         {
-            SLOTTYPE slot = SLOT_MAIN;
-
-            if (m_attackDirection == PHYSICAL_ATTACK_DIRECTION::RIGHTATTACK)
+            if (auto* weapon = dynamic_cast<CItemWeapon*>(m_attacker->m_Weapons[weaponSlot]))
             {
-                slot = SLOT_SUB;
+                skilltype = static_cast<SKILLTYPE>(weapon->getSkillType());
```

Comment:
Why is getSkillType not returning SKILLTYPE anyway?

Reply:
> Why is getSkillType not returning SKILLTYPE anyway?

Yeah I am not sure, `GetWeaponSlot` also returns a uint8 for some reason. I can fix the functions in a separate PR if we want?